File: source_code - v1/FUNC_Create_SAT_UE.py
import numpy as np


def FUNC_Create_SAT_UE(Area, numSAT, numUE, zm):
    x_area = Area[0]
    y_area = Area[1]

    sat = np.zeros((3, 4), dtype=float)
    sat[0, :] = [x_area / 4, x_area * 3 / 4, x_area * 3 / 4, x_area / 4]
    sat[1, :] = [y_area / 4, y_area / 4, y_area * 3 / 4, y_area * 3 / 4]
    sat[2, :] = [zm, zm, zm, zm]

    ue = np.zeros((3, numUE), dtype=float)
    ue[0, :] = x_area * np.random.rand(numUE)
    ue[1, :] = y_area * np.random.rand(numUE)
    return sat, ue

File: source_code - v1/Test_convergence/FUNC_Create_SAT_UE.m
function [SAT, UE] = FUNC_Create_SAT_UE(Area, numSAT, numUE, zm)
x_area = Area(1); y_area = Area(2);

%% Create SATs
% Firstly, assume numSAT = 4
SAT = zeros(3, 4);
SAT(1,:) = [x_area/4 x_area*3/4 x_area*3/4 x_area/4];
SAT(2,:) = [y_area/4 y_area/4 y_area*3/4 y_area*3/4];
SAT(3,:) = [zm zm zm zm];

%% Create UEs
UE = zeros(3, numUE);
UE(1,:) = x_area*rand(1,numUE);
UE(2,:) = y_area*rand(1,numUE);

end